raiwidgets/raiwidgets/_cohort.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Module for defining cohorts in raiwidgets package."""
+
+from typing import Any, List
+
+import pandas as pd
+
+from responsibleai.exceptions import UserConfigValidationException
+
+
+class CohortFilterMethods:
+    """Defines different methods for cohort filters.
+    """
+    GreaterThan = "greater"  # Strictly greater than a value
+    # Greater than or equal to a value
+    GreaterThanEqualTo = "greater and equal"
+    LessThan = "less"  # Strictly lesser than a value
+    LessThanEqualTo = "less and equal"  # Lesser than or equal to a value
+    Equal = "equal"  # Equal to a value
+    Includes = "includes"  # Includes a set of values
+    Excludes = "excludes"  # Excludes a set of values
+    InTheRangeOf = "in the range of"  # In a given range of two values
+
+    # List of all filter methods
+    ALL = [GreaterThan,
+           GreaterThanEqualTo,
+           LessThan,
+           LessThanEqualTo,
+           Equal,
+           Includes,
+           Excludes,
+           InTheRangeOf]
+
+    # List of filter methods which expect a single value from the user
+    SINGLE_VALUE_METHODS = [GreaterThanEqualTo,
+                            GreaterThan,
+                            LessThanEqualTo,
+                            LessThan,
+                            Equal]
+
+
+def cohort_filter_json_converter(obj):
+    """Helper function to convert CohortFilter object to json.
+
+    :param obj: Object to convert to json.
+    :type obj: object
+    :return: The converted json.
+    :rtype: dict
+    """
+    if isinstance(obj, CohortFilter):
+        return obj.__dict__
+    try:
+        return obj.to_json()
+    except AttributeError:
+        return obj.__dict__
+
+
+class CohortFilter:
+    """Defines the cohort filter.
+
+    :param method: Cohort filter method from one of CohortFilterMethods.
+    :type method: str
+    :param arg: List of values to be used by the cohort filter.
+    :type arg: list
+    :param column: The column name from the dataset on which the filter
+                   will be applied.
+    :type column: str
+    """
+    def __init__(self, method: str, arg: List[Any], column: str):
```

Comment:
it looks like this doesn't support composite filter, but maybe that's ok for the initial version

Reply:
The cohort defined from the UI (not from error analysis)  is not of type Composite filter. So on similar lines keeping the initial implementation simple to just include filter.